File: Search specificly/WindowsFormsApp1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Data.SqlClient;

namespace WindowsFormsApp1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        SqlConnection con = new SqlConnection("Data Source=(LocalDB)\\MSSQLLocalDB;AttachDbFilename=|DataDirectory|\\location.mdf;Integrated Security=True");
        private void Form1_Load(object sender, EventArgs e)
        {
            // TODO: This line of code loads data into the 'locationDataSet.Table' table. You can move, or remove it, as needed.
            this.tableTableAdapter.Fill(this.locationDataSet.Table);
            //comboBox1.SelectedIndex = 1;
        }

        private void textBox1_KeyDown(object sender, KeyEventArgs e)
        {
            //if (e.KeyCode == Keys.Enter)
            //{
            //    if (string.IsNullOrEmpty(textBox1.Text))
            //        tableBindingSource.Filter = string.Empty;
            //    else
            //        tableBindingSource.Filter = string.Format("{0}='{1}'", comboBox1.Text, textBox1.Text);


            //}
        }

        private void textBox1_TextChanged(object sender, EventArgs e)
        {
            if (comboBox1.Text == "Id")
            {
                SqlDataAdapter sda = new SqlDataAdapter("Select Id, Street, Suburb, City FROM [Table] WHERE Id LIKE '%" + textBox1.Text+"%'",con);
                DataTable data = new DataTable();
                sda.Fill(data);
                dataGridView1.DataSource = data;
            }
            //else if (comboBox1.Text == "Place")
            //{
            //    SqlDataAdapter sda = new SqlDataAdapter("Select Id, Place, Street, Suburb, City FROM [Table] WHERE Place LIKE '%" + textBox1.Text + "%'", con);
            //    DataTable data = new DataTable();
            //    sda.Fill(data);
            //    dataGridView1.DataSource = data;
            //}
            else if (comboBox1.Text == "Street")
            {
                SqlDataAdapter sda = new SqlDataAdapter("Select Id, Street, Suburb, City FROM [Table] WHERE Street LIKE '%" + textBox1.Text + "%'", con);
                DataTable data = new DataTable();
                sda.Fill(data);
                dataGridView1.DataSource = data;
            }
            else if (comboBox1.Text == "Suburb")
            {
                SqlDataAdapter sda = new SqlDataAdapter("Select Id, Street, Suburb, City FROM [Table] WHERE Suburb LIKE '%" + textBox1.Text + "%'", con);
                DataTable data = new DataTable();
                sda.Fill(data);
                dataGridView1.DataSource = data;
            }
            else if (comboBox1.Text == "City")
            {
                SqlDataAdapter sda = new SqlDataAdapter("Select Id, Street, Suburb, City FROM [Table] WHERE City LIKE '%" + textBox1.Text + "%'", con);
                DataTable data = new DataTable();
                sda.Fill(data);
                dataGridView1.DataSource = data;
            }
        }
    }
}

File: Duplicate sprint3/add/Sign Meng Meng/Form1.vb
Imports System.Data.SqlClient

Public Class Form1
    Dim con As New SqlConnection
    Dim cmd As New SqlCommand

    Private Sub Form1_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        'TODO: This line of code loads data into the 'DataSet.table1' table. You can move, or remove it, as needed.
        Me.Table1TableAdapter.Fill(Me.DataSet.table1)

        con.ConnectionString = "Data Source=(LocalDB)\MSSQLLocalDB;AttachDbFilename=|DataDirectory|\Location.mdf;Integrated Security=True"
        If con.State = ConnectionState.Open Then
            con.Close()

        End If
        con.Open()

        disp_data()
    End Sub

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click

        cmd = con.CreateCommand()
        cmd.CommandType = CommandType.Text
        cmd.CommandText = "insert into table1 values('" + TextBox1.Text + "','" + TextBox2.Text + "','" + TextBox3.Text + "','" + TextBox4.Text + "')"
        cmd.ExecuteNonQuery()

        MessageBox.Show("Record Inserted Successfully")
        disp_data()

    End Sub

    Private Sub disp_data()

        cmd = con.CreateCommand()
        cmd.CommandType = CommandType.Text
        cmd.CommandText = "select * from table1"
        cmd.ExecuteNonQuery()
        Dim dt As New DataTable()
        Dim da As New SqlDataAdapter(cmd)
        da.Fill(dt)
        DataGridView1.DataSource = dt

    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        cmd = con.CreateCommand()
        cmd.CommandType = CommandType.Text
        cmd.CommandText = "WITH CTE AS
(
SELECT *,ROW_NUMBER() OVER (PARTITION BY Street,Suburb,City ORDER BY Street,Suburb,City) AS RN
FROM table1
)

DELETE FROM CTE WHERE RN<>1"

        '"DELETE FROM table1 WHERE Street  In
        ''(SELECT Street FROM table1
        ''WHERE street = 'East')"







        cmd.ExecuteNonQuery()

        MessageBox.Show("Record Deleted Successfully")
        disp_data()
    End Sub
End Class
